var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -180,6 +180,26 @@ def cmake_args(self):
         spec = self.spec
         args = []
 
+        # This bit of code is needed in order to pass compiler.yaml flags
+        # into the QMCPACK's CMake. Probably the CMake base class in
+        # the code of Spack should be doing this instead. Otherwise, it
+        # it would need to be done on a per package basis which is
+        # problematic.
+        cflags = spec.compiler_flags['cflags']
+        cxxflags = spec.compiler_flags['cxxflags']
+        args.append('-DCMAKE_C_FLAGS=%s' % ' '.join(cflags))
+        args.append('-DCMAKE_CXX_FLAGS=%s' % ' '.join(cxxflags))
+
+        # This issue appears specifically with the the Intel compiler,
+        # but may be an issue with other compilers as well. The final fix
+        # probably needs to go into QMCPACK's CMake instead of in Spack.
+        # QMCPACK binaries are linked with the C++ compiler, but *may* contain
+        # Fortran libraries such as NETLIB-LAPACK and OpenBLAS on the link
+        # line. For the case of the Intel C++ compiler, we need to manually
+        # add a libray from the Intel Fortran compiler.
+        if self.compiler.name == 'intel':
```

Comment:
I think:
```python
if '%intel' in self.spec:
    ...
```
is more idiomatic. @adamjstewart What do you think?

Reply:
If the dependency on `libifcore` is not direct, but just comes transitively through libraries, it might be the case that the Intel compiler is not setting RPATH correctly on those libraries. Just telling in case you want to double check.

Reply:
> I think:
> 
> ```python
> if '%intel' in self.spec:
>     ...
> ```
> 
> is more idiomatic. @adamjstewart What do you think?

I can change it to '%intel' -- there is more than one way to do things in Spack and I usually end up just copying what I see in another package (either for bad or for good).

Reply:
> If the dependency on `libifcore` is not direct, but just comes transitively through libraries, it might be the case that the Intel compiler is not setting RPATH correctly on those libraries. Just telling in case you want to double check.

So this problem occurs at compile time even with static linking. So, I don't think its an RPATH issue. I even have the appropriate directory set via EXTRA_RPATH in `packages.yaml.` As an aside issue, EXTRA_RPATH is also not automatically passed into CMake as far as I can tell either.